### ISSUE TITLE
When I use SeparableConv2D(padding =same) and change the 'kernel_size' and 'strides', the different backends may return different output, theano even can return a different shape output

### ISSUE THREAD

**Issue by Jup11ter** [keras-team/keras#13849]
## System information

- Have I written custom code (as opposed to using example directory):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 & Linux Ubuntu 18.04
- Tensorflow backend (yes / no): yes
- Tensorflow version：1.14.0-gpu
- Cntk version: 2.7-gpu
- Theano version:1.04-gpu
- Keras version: 2.3.1
- Python version: 3.6.9
- CUDA/cuDNN version: -
- GPU model and memory: -

## Describe the current behavior
**When I use `SeparableConv2D(padding=same)` , and change the `kernel_size` and `strides`, I notice obvious output differences, and even different output shapes across 3 backends in some cases** as shown in the following tables.  (See the codes below to reproduce the issue.)

Note that, **we take the sum of absolute diffenece (SoAD) as the metric** in these tables. (See SoAD implementation in code snippet below.)

1) `kernel_size=3`

**Three backends do not show definitive similarity facing different `strides` values.** 

|strides|Tensorflow-CNTK|Tensorflow-Theano|CNTK-Theano|
| ----: | ---: |---:| ---------: |
|1|3.75624584e-12|4.25373454e-12 |3.61926866e-12 |
|2|**3.85814342e+03**|  **3.85814342e+03**|8.29253658e-13|
|3|**1.69542644e+03**|**1.69542644e+03**| 3.25148545e-13 |
|4|2.49435889e-13|**9.87705031e+02**| **9.87705031e+02**|
|5|**7.75239298e+02**| **7.75239298e+02**| 1.57913613e-13|
|6|**6.29568065e+02**|**6.29568065e+02** |1.24693658e-13 |
|7|1.02974920e-13|**3.85866109e+02**|**3.85866109e+02**|
|8|**2.51915201e+02**| **2.65104737e+02** | **2.67472971e+02** |
|9|**2.26283740e+02**|**2.33788407e+02**| **2.50488568e+02** |
|10|2.25733297e+02|2.25733297e+02| 4.43083070e-14|

2) `kernel_size=2`

Output inconsistencies occur across backends, just similar to 1). 

And **surprisingly, when  `kernel_size=2` and `strides=1/2/4/8`, Theano's outputs have different shapes with the outputs of Tensorflow or CNTK , **  as shown in the table below.

|strides|Tensorflow-CNTK|Tensorflow-Theano|CNTK-Theano|
| ----: | ---: |---:| ---------: |
|***1***|***13641.496737474721***|-| -|
|***2***|8.257352592487555e-13| -|-|
|3|4.280985288485084e-13| 1761.6978802073534 |1761.6978802073534|
|***4***|***825.0764765572205***|-|-|
|5|1.0748736969934392e-13|619.0982105268306|619.0982105268306|
|6|1.163895368971879e-13|523.277989721278|523.277989721278|
|7|343.929670036006|452.48023212786035|393.4513363667048|
|***8***|***242.51455245634935***|-|-|
|9|229.78542172957555|262.94141575947594|268.8390808614687|
|10|4.9367628041085965e-14|232.66676896258892|232.66676896258892|

**Shape comparison when `strides=1/2/4/8`** 

| strides|Tensorflow shape|CNTK shape|Theano shape|
|-:|-:|-:|-:|
|1| (1, 32, 32, 12) | (1, 32, 32, 12) |(1, 33, 33, 12)|
|2| (1, 16, 16, 12)| (1, 16, 16, 12) |(1, 17, 17, 12)|
| 4| (1, 8, 8, 12)| (1, 8, 8, 12) |(1, 9, 9, 12)|
|8|(1, 8, 8, 12)|(1, 8, 8, 12)|(1, 5, 5, 12)|

## Key insights

1. It is explictly mentioned in `Conv2D`'s docs that ***Note that `"same"` is slightly inconsistent across backends with`strides` != 1***,(see `keras\layers\convolutional.py line391` ). **This also applies to the case of `SeparableConv2D`,  but we found no similar warnings for `SeparableConv2D`**. I think Keras needs to update its comments here. 

2. Worse still, even when `strides=1` (e.g., `kernel_size=2` and `strides=1`), there shows significant output difference between CNTK and Tensorflow.  

3. **Theano even generates different output shapes with other two backends in some cases** (e.g.,  `kernel_size=2 ,stride = 1/2/4/8`).  Note that, this problem only occurs when `padding=same` in `SeparableCon2D`.  For other convolution layers like `Conv2D`,  the output shape always keep the same across the 3 backends,  although there are value differences in some senarios.

   **Dose it indicate a Theano implementation bug in `SeparableConv2D`?  This may lead to potential risk when we use `SeparableConv2D` in real applications.**

## Code to reproduce the issue

```python
import os
import numpy as np
import keras.layers as L
import keras.backend as K
import importlib
from keras.models import load_model
from keras.engine import Model, Input

backends = ['cntk','tensorflow','theano']

# SoAD, Sum of Absolute Difference
def acc_abs_diff(output1, output2):
    assert output1.shape == output2.shape
    abs_diff = np.abs(output1-output2)
    return  np.sum(abs_diff)

def set_keras_backend(backend='tensorflow'):
    if K.backend() != backend:
        os.environ['KERAS_BACKEND'] = backend
        importlib.reload(K.load_backend)
        importlib.reload(K)
        assert K.backend() == backend


kwargs={'filters':12,'kernel_size': 2, 'padding': 'same', 'strides': 1, 'dilation_rate': (1, 1), 'data_format': 'channels_last'}
#kwargs['filters']=12 #you may use 'filter' in the tests 
listdiff=[]
for stride_value in range(10):
    kwargs['strides']=stride_value+1
    input_data = (10 * np.random.random((1,32,32,16)))
    input = input_data.astype('float64')
    set_keras_backend('tensorflow')
    layer = L.convolutional.SeparableConv2D(**kwargs)# you can also use Conv2D
    x = Input(batch_shape=input.shape)
    y = layer(x)
    bk_model = Model(x, y)
    model_path = os.path.join('./', 'model.h5')
    bk_model.save(model_path, bk_model)
    output = {}
    for bk in backends:
        try:
            set_keras_backend(backend=bk)
            model = load_model(model_path)
            output[bk] = model.predict(input)
        except:
            print('error result')
    try:
        diff1 = acc_abs_diff(output['tensorflow'], output['cntk'])
    except:
        diff1 = None
    try:
        diff2=acc_abs_diff(output['theano'], output['cntk'])
    except:
        diff2=None
    try:
        diff3=acc_abs_diff(output['theano'], output['tensorflow'])
    except:
        diff3=None
        print([output['theano'].shape,output['tensorflow'].shape,kwargs['strides']])
    listdiff.append([diff1,diff2,diff3])
arraydiff = np.array(listdiff)
print(arraydiff)
print('finish')
```

**Comment by Saduf2019** [comment]
@Jup11ter 
We see that you are using old version of tensorflow which is not actively supported now, please verify on 2.x and open a new issue in case you face any errors, hence moving this to closed status.
